src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,144 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (size == 0) {
+            newNode = new Node<>(null, value, null);
+            head = newNode;
+        } else if (size == 1) {
+            newNode = new Node<>(head, value, null);
+            head.next = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+        if (index == size) {
+            add(value);
+        } else {
+            if (index == 0) {
+                newNode = new Node<>(null, value, head);
+                head.prev = newNode;
+                head = newNode;
+            } else {
+                checkIndex(index);
+                Node<T> prevNode = getNode(index - 1);
+                Node<T> nextNode = getNode(index);
```

Comment:
call **getNode** once
```suggestion
                Node<T> nextNode = getNode(index);
```